src/routes/v1/getTxInfo.ts

```diff
@@ -50,62 +57,81 @@ getTxInfoRouter.get('/', async function(req, res) {
     }
   )
 
-  if (query.addressPrefix.length < 3) {
-    res.status(400).send('addressPrefix must be at least 3 characters')
+  if (query.addressHashfix < 0 || query.addressHashfix > 2 ** 40) {
+    res.status(400).send('addressHashfix must be between 0 and 2^40')
     return
   }
 
   const startDate = new Date(query.startIsoDate)
   const endDate = new Date(query.endIsoDate)
+  const startKey = [query.addressHashfix, startDate.toISOString()]
```

Comment:
start and end have the same hash. This doesn't allow for any privacy to ensure we don't exactly know the users addresss.

If this were a standard sha256 hash into hex, we could use the first N chars as the prefix to allow for privacy

Reply:
The hashfix is not collision resistant. It is a "hash prefix" because it takes 5 bytes from the address (5 bytes is 5 chars).

Reply:
OTS comments:

1. Use a large enough space size of (I recall 64^6)
2. Use mango queries to do range queries on the hashfix value

The goal is to be flexible in our design. We want to query different collision space over time rather than be locked to some byte size. To do this we need to do range queries on the hashfix. So the space for the hashfix is a maximum, but the range will determine the actual collision space.